webpack.config.js

```diff
@@ -1,30 +1,52 @@
 const path = require('path');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const OptimizeJsPlugin = require('optimize-js-plugin');
+const plugins = [
+  new HtmlWebpackPlugin({
+    template: './src/index.html',
+    filename: 'index.html',
+    inject: 'body'
+  })
+];
 
-module.exports = (env) => ({
-  mode: env || 'production',
-  entry: './src/index.js',
-  output: {
-    path: path.resolve(__dirname, 'build'),
-    filename: 'app.bundle.js'
-  },
-  module: {
-    rules: [
-      {
-        test: /\.js$/,
-        loader: 'babel-loader'
-      },
-      {
-        test: /\.css$/,
-        use: [
-          { loader: 'style-loader'},
-          {
-            loader: 'css-loader',
-            options: {
-              modules: true
-            }
-          }
-        ]
-      }
-    ]
+module.exports = env => {
+  if (env === 'production') {
+    plugins.push(
+      new OptimizeJsPlugin({
+        sourceMap: false
+      })
+    );
   }
-});
\ No newline at end of file
+  return {
+    mode: env || 'production',
+    entry: './src/index.js',
+    output: {
+      path: path.resolve(__dirname, 'build'),
+      filename: 'app.bundle.js'
+    },
+    module: {
+      rules: [
+        {
+          test: /\.js$/,
+          loader: 'babel-loader',
+          options: {
+            plugins: env !== 'production' ? ['react-hot-loader/babel'] : []
+          }
+        },
+        {
+          test: /\.css$/,
+          use: [
+            { loader: 'style-loader' },
+            {
+              loader: 'css-loader',
+              options: {
+                modules: true
+              }
+            }
+          ]
+        }
+      ]
+    },
+    plugins: plugins
```

Comment:
if value and key are the same you can write just one of them, in this case just `plugins`